pkg/cli/mirror/mirror.go

```diff
@@ -720,7 +721,7 @@ func (o *MirrorOptions) mirrorToMirrorWrapper(ctx context.Context, cfg v1alpha2.
 			return err
 		}
 		workspace := filepath.Join(o.Dir, config.SourceDir)
-		if err = metadata.UpdateMetadata(ctx, sourceBackend, &meta, workspace, o.SourceSkipTLS, o.SourcePlainHTTP); err != nil {
+		if err = metadata.UpdateMetadata(ctx, sourceBackend, &meta, workspace, o.SourceSkipTLS, o.SourcePlainHTTP, allCatalogs); err != nil {
```

Comment:
Since we're passing the map to UpdateMetadata, we should check that the workflow MirrorToDisk + DiskToMirror is not impacted by this change.
If it is working, we can even extend this nice idea, and pass the map to rebuildCatalogs, thus completely bypassing the necessity to walk the folder to "discover" the catalogs that need rebuilding.